components/connector/internal/certificates/service.go

```diff
@@ -92,7 +92,7 @@ func (svc *certificateService) encodeCertificates(rawCaCertificate, rawClientCer
 			return EncodedCertificateChain{}, err
 		}
 
-		caCrtBytes = svc.createCertChain(rootCABytes, caCrtBytes)
+		caCrtBytes = svc.createCertChain(caCrtBytes, rootCABytes)
 	}
```

Comment:
Very small suggestion (you can safely ignore it). Maybe it could be be worthy to rename createCertChain function and its arguments to reflect that it adds next derived certificate into the chain? For example appendNewDerivedCertificate(derivedCrt, chain []byte).